ui/src/criteria/attribute.tsx

```diff
@@ -138,8 +286,17 @@ type ConceptDetailsProps = {
 function ConceptDetails(props: ConceptDetailsProps) {
   return (
     <>
-      {props.data.selected.length === 0 ? (
+      {props.data.selected.length === 0 &&
```

Comment:
Hi Tim, It could not correct show the content while I use the if in the typescript HTML. Maybe I should keep it here.

Reply:
I believe this is because you were initializing both `selected` and `min/max` at the same time combined with however you were writing your if statements.

In future, when you encounter something that doesn't work the way you expect you should try to understand why. There's no obvious reason why an if statement wouldn't work the same as the ternary operator so you should look at how the code is different. I would guess the difference is that you split the `selected.length` and `isValid(min/max)` parts into separate if statements and it always displayed the enum data, which makes sense given the initialization issue. You could try using the debugger or `console.log` to inspect the state.

Reply:
Ok, I over complicated the problem before... I thought I need to put the if statement into the HTML part, something like:
```
return (
     <> 
          {if (isValid(XXX)) {} else {}}
     </>
)
```
which of course doesn't work. Now I put the return statement into the if statement and make it looks simpler. Thanks for your advice!